sdk/python/tests/README.md

```diff
@@ -0,0 +1,341 @@
+            "conv_rate_plus_100:conv_rate_plus_100_rounded",
+            "conv_rate_plus_100:conv_rate_plus_val_to_add",
+            "order:order_is_success",
+            "global_stats:num_rides",
+            "global_stats:avg_ride_length",
+            "field_mapping:feature_name",
+        ],
+        full_feature_names=full_feature_names,
+    )
+
+    if job_from_df.supports_remote_storage_export():
+        files = job_from_df.to_remote_storage()
+        print(files)
+        assert len(files) > 0  # This test should be way more detailed
+
+    start_time = datetime.utcnow()
+    actual_df_from_df_entities = job_from_df.to_df()
+    # ... more test code
+
+    validate_dataframes(
+        expected_df,
+        table_from_df_entities,
+        keys=[event_timestamp, "order_id", "driver_id", "customer_id"],
+    )
+    # ... more test code
+```
+{% endtab %}
+{% endtabs %}
+
+- The key fixtures are the `environment` and `universal_data_sources` fixtures, which are defined in the `feature_repos` directories and the `conftest.py` file. This by default pulls in a standard dataset with driver and customer entities(that we have pre-defined), certain feature views, and feature values.
+    - The `environment` fixture sets up a feature store, parametrized by the provider and the online/offline store. It allows the test to query against that feature store without needing to worry about the underlying implementation or any setup that may be involved in creating instances of these datastores.
+    - Each fixture creates a different integration test with its own `IntegrationTestRepoConfig` which is used by pytest to generate a unique test testing one of the different environments that require testing.
+
+- Feast tests also use a variety of markers:
+    - The `pytest.mark.integration` marker is used to designate integration tests which will cause the test to be run when you call `make test-python-integration`.
+    - The `@pytest.mark.universal_offline_stores` marker will parametrize the test on all of the universal offline stores including file, redshift, bigquery and snowflake.
+    - The `full_feature_names` parametrization defines whether or not the test should reference features as their full feature name(fully qualified path) or just the feature name itself.
+
+
+## Writing a new test or reusing existing tests
+
+### To add a new test to an existing test file
+
+* Use the same function signatures as an existing test (e.g. use `environment` and `universal_data_sources` as an argument) to include the relevant test fixtures.
+* If possible, expand an individual test instead of writing a new test, due to the cost of starting up offline / online stores.
+* Use the `universal_offline_stores` and `universal_online_store` markers to parametrize the test against different offline store and online store combinations. You can also designate specific online and offline stores to test by using the `only` parameter on the marker.
```

Comment:
maybe add an example of how to use `only`, since it's nonobvious?

Reply:
fix